docs/ALGORITHM.md

```diff
@@ -0,0 +1,91 @@
+# Finality Quorum Calculation Algorithm
+
+## Table of Contents
+
+1. [Overview](#1-overview)  
+2. [Inputs and Data Sources](#2-inputs-and-data-sources)  
+3. [Step‑by‑Step Algorithm](#3-step-by-step-algorithm)
+
+---
+
+## 1. Overview
+
+The Finality Gadget determines if a rollup block is finalized by 
+checking if a quorum of voting power has signed the block. This 
+involves fetching votes from the contract, calculating each 
+provider’s voting power, and comparing the total voted power to 
+the required threshold (> 2/3).
+
+---
+
+## 2. Inputs and Data Sources
+
+1. Rollup blocks  
+   - Fetch height, hash, and timestamp from the rollup chain  
+2. Finality Providers  
+   - Fetch FP public keys from the Babylon chain  
+3. Voting Power  
+   - Calculate FP voting power via Bitcoin delegation data at the 
+     specific BTC block height  
+4. Quorum Threshold  
+   - Hardcoded as 2/3 of total voting power  
```

Comment:
I think we can divide this section into two, one for data source and the other for input. 

- The data source include Babylon Genesis (and the rollup contract), a rollup full node, and a Bitcoin full node.
- The input is only a single rollup block, and the algorithm would have a signature like `func is_btc_staking_finalized(block)`. It is finality gadget's responsibility to run this algorithm for every block. Within this list, 
  - Finality providers and their voting power are fetched from Babylon through the given rollup block
  - Quorum threshold 2/3 is a constant